lib/src/realm_class.dart

```diff
@@ -474,3 +464,74 @@ class RealmLogLevel {
   /// Same as [Level.OFF];
   static const off = Level.OFF;
 }
+
+/// @nodoc
+class RealmMetadata {
+  final Map<Type, RealmObjectMetadata> _typeMap = <Type, RealmObjectMetadata>{};
+  final Map<String, RealmObjectMetadata> _stringMap = <String, RealmObjectMetadata>{};
+
+  RealmMetadata._(Iterable<RealmObjectMetadata> objects) {
+    for (final meta in objects) {
+      if (meta.type != RealmObject) {
+        _typeMap[meta.type] = meta;
+      } else {
+        _stringMap[meta.name] = meta;
+      }
+    }
+  }
+
+  RealmObjectMetadata getByType(Type type) {
+    final metadata = _typeMap[type];
+    if (metadata == null) {
+      throw RealmError("Object type $type not configured in the current Realm's schema. Add type $type to your config before opening the Realm");
+    }
+
+    return metadata;
+  }
+
+  RealmObjectMetadata getByName(String type) {
```

Comment:
How is this going to behave if there are same type names in different packages or libraries?

Reply:
We don't support tables with the same name, so that would have been an error when opening the Realm file.